rust/blockstore/src/arrow/block/delta/ordered_block_delta.rs

```diff
@@ -113,38 +100,43 @@ impl OrderedBlockDelta {
     }
 
     fn copy_up_to<'a, K: ArrowReadableKey<'a>, V: ArrowReadableValue<'a>>(
-        &mut self,
+        &'a mut self,
         excluded_prefix: &str,
-        excluded_key: KeyWrapper,
-        old_block: &'a Block,
+        excluded_key: &KeyWrapper,
     ) {
-        let prefix_arr = old_block
-            .data
-            .column(0)
-            .as_any()
-            .downcast_ref::<StringArray>()
-            .unwrap();
-        let key_arr = old_block.data.column(1);
-
-        for i in self.copied_up_to_row_of_old_block..old_block.data.num_rows() {
-            let old_prefix = prefix_arr.value(i);
-            let old_key = K::get(key_arr, i);
-            let old_key_wrapped: KeyWrapper = old_key.clone().into(); // todo: remove clone
-
-            if old_prefix > excluded_prefix
-                || (old_prefix == excluded_prefix && old_key_wrapped >= excluded_key)
-            {
-                // If the provided prefix/key pair existed in the old block, we need to skip it (move the pointer past that entry)
-                if old_prefix == excluded_prefix && old_key_wrapped == excluded_key {
-                    self.copied_up_to_row_of_old_block += 1;
+        if let Some(old_block) = self.old_block.as_ref() {
+            let prefix_arr = old_block
+                .data
+                .column(0)
+                .as_any()
+                .downcast_ref::<StringArray>()
+                .unwrap();
+            let key_arr = old_block.data.column(1);
+
+            for i in self.copied_up_to_row_of_old_block..old_block.data.num_rows() {
+                let old_prefix = prefix_arr.value(i);
+                let old_key = K::get(key_arr, i);
+
+                match old_prefix.cmp(excluded_prefix) {
+                    std::cmp::Ordering::Less => {}
+                    std::cmp::Ordering::Equal => {
+                        match old_key.clone().into().partial_cmp(excluded_key) {
+                            Some(std::cmp::Ordering::Less) => {}
+                            Some(std::cmp::Ordering::Equal) => {
+                                self.copied_up_to_row_of_old_block += 1;
+                                break;
+                            }
+                            Some(std::cmp::Ordering::Greater) => break,
+                            None => panic!("Could not compare keys"),
```

Comment:
nervous about introducing a panic here. Is the reason to switch to cmp api just cleanliness?

Reply:
this performs at most 1 compare of the prefix and 1 compare of the key
previously the prefix could be compared up to 3 times and the key could be compared up to 2 times
the only reason it would panic is if two f32s were not comparable 
we make the assumption a few other places that f32 keys are always comparable, and panic if they don't, so I figured it was ok to do it here
maybe it would make more sense to have `KeyWrapper` implement `Ord` instead of just `PartialOrd`?

Reply:
Can we use `f32::total_cmp`?  It makes your assumption true.

Reply:
discussed offline with @rescrv, will leave as-is for now and update in the future